environment/config.go

```diff
@@ -95,112 +134,135 @@ type ConfigureResults struct {
 
 // ConfigurationParameters are options used to create new ConfigureResults
 type ConfigurationParameters struct {
-	InstrumentOpts instrument.Options
-	HashingSeed    uint32
-	HostID         string
+	InstrumentOpts             instrument.Options
+	HashingSeed                uint32
+	HostID                     string
+	NamespaceResolutionTimeout time.Duration
 }
 
 // Configure creates a new ConfigureResults
 func (c Configuration) Configure(cfgParams ConfigurationParameters) (ConfigureResults, error) {
+	var emptyConfig ConfigureResults
+
+	if c.Service.SDConfig.InitTimeout == 0 {
+		c.Service.SDConfig.InitTimeout = defaultSDTimeout
+	}
+
+	configSvcClientOpts := c.Service.NewOptions().
+		SetInstrumentOptions(cfgParams.InstrumentOpts).
+		SetServiceDiscoveryConfig(c.Service.SDConfig)
+	configSvcClient, err := etcdclient.NewConfigServiceClient(configSvcClientOpts)
+	if err != nil {
+		err = fmt.Errorf("could not create m3cluster client: %v", err)
+		return emptyConfig, err
+	}
+
+	if c.Service != nil && c.Static != nil {
+		return emptyConfig, errInvalidConfig
+	}
+
+	if c.Service != nil {
+		return c.configureDynamic(configSvcClient, cfgParams)
+	}
+
+	if c.Static != nil {
+		return c.configureStatic(configSvcClient, cfgParams)
+	}
+
+	return emptyConfig, errInvalidConfig
+}
 
+func (c Configuration) configureDynamic(configSvcClient client.Client, cfgParams ConfigurationParameters) (ConfigureResults, error) {
+	dynamicOpts := namespace.NewDynamicOptions().
+		SetInstrumentOptions(cfgParams.InstrumentOpts).
+		SetConfigServiceClient(configSvcClient).
+		SetNamespaceRegistryKey(kvconfig.NamespacesKey).
+		SetInitTimeout(cfgParams.NamespaceResolutionTimeout)
```

Comment:
I was able to set up a cluster of seed nodes talking to each other, although the flat duration wait does indeed make you "race" against the timeout duration. When the m3db node is started, it first sets up the embedded etcd (if it's a seed node) and the topology initializer will wait (block) the specified duration for a topology, during which you can issue your setup commands.

Reply:
Ah I see yeah.  Ok great, would be good to get this changed later (once this has landed) to avoid the need to race against the clock.  Good to know it doesn't block etcd startup.